### PR TITLE
Clean up after transition into opaque pointers. NFC

### PR DESCRIPTION
**Pull request by bjope**
A couple of minor clean up commits.

### CODE REVIEW

File: llvm/examples/ExceptionDemo/ExceptionDemo.cpp [hunk 1]
```diff
@@ -1244,8 +1244,7 @@ static llvm::Function *createCatchWrappedInvokeFunction(
   llvm::Value *unwindExceptionClass =
       builder.CreateLoad(builder.CreateStructGEP(
           ourUnwindExceptionType,
-          builder.CreatePointerCast(unwindException,
-                                    ourUnwindExceptionType->getPointerTo()),
+          builder.CreatePointerCast(unwindException, builder.getPtrTy()),
```
Comment (nikic) [inline llvm/examples/ExceptionDemo/ExceptionDemo.cpp:1247]: Also drop the pointer cast.

File: llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp [hunk 2]
```diff
@@ -4567,7 +4567,7 @@ Instruction *InstCombinerImpl::foldNot(BinaryOperator &I) {
     Type *SextTy = cast<BitCastOperator>(NotOp)->getSrcTy();
     Value *NotX = Builder.CreateNot(X);
     Value *Sext = Builder.CreateSExt(NotX, SextTy);
-    return CastInst::CreateBitOrPointerCast(Sext, Ty);
+    return CastInst::Create(Instruction::BitCast, Sext, Ty);
```
Comment (nikic) [inline llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp:4570]:
```suggestion
    return new BitCastInst(Sext, Ty);
```

File: llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp [hunk 3]
```diff
@@ -2386,7 +2386,7 @@ static Instruction *foldSelectCmpBitcasts(SelectInst &Sel,
   } else {
     return nullptr;
   }
-  return CastInst::CreateBitOrPointerCast(NewSel, Sel.getType());
+  return CastInst::Create(Instruction::BitCast, NewSel, Sel.getType());
```
Comment (nikic) [inline llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp:2389]:
```suggestion
  return new BitCastInst(NewSel, Sel.getType());
```

File: llvm/examples/ExceptionDemo/ExceptionDemo.cpp (hunk 1 above)
Comment (bjope) [reply llvm/examples/ExceptionDemo/ExceptionDemo.cpp]: Done!

File: llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp (hunk 2 above)
Comment (bjope) [reply llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp]: Done!

File: llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp (hunk 3 above)
Comment (bjope) [reply llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp]: Done!

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**